@here/harp-mapview/lib/geometry/SolidLineMesh.ts

```diff
@@ -0,0 +1,451 @@
+    DisplacementRange,
+    expandBoxByDisplacementRange
+} from "./DisplacedBufferGeometry";
+
+const tmpSphere = new THREE.Sphere();
+const tmpInverseMatrix = new THREE.Matrix4();
+const tmpRay = new THREE.Ray();
+const tmpLine1 = new THREE.Line3();
+const tmpBox = new THREE.Box3();
+const tmpPlane = new THREE.Plane();
+const tmpV1 = new THREE.Vector3();
+const tmpV2 = new THREE.Vector3();
+const tmpV3 = new THREE.Vector3();
+const tmpV4 = new THREE.Vector3();
+
+// Strides to access the index buffer. See [[createLineGeometry]].
+// Stride between the start vertex indices of consecutive segments, each one made of 2 triangles.
+const SEGMENT_STRIDE = 6;
+// Stride between the start and end vertex indices of a segment. Vertices are duplicated so that
+// each copy is extruded in opposite directions in the vertex shader.
+const VERTEX_STRIDE = 2;
+
+function isSolidLineMaterial(material: THREE.Material | THREE.Material[]): boolean {
+    return Array.isArray(material)
+        ? material.every(mat => mat instanceof SolidLineMaterial)
+        : material instanceof SolidLineMaterial;
+}
+
+/**
+ * Computes the bounding sphere of the part of a given geometry corresponding to a feature.
+ * @param geometry The geometry containing the feature.
+ * @param featureBeginIndex The index where the feature starts in the geometry's indices attribute.
+ * @param featureEndIndex The index where the feature end in the geometry's indices attribute.
+ * @returns The feature bounding sphere.
+ */
+function computeFeatureBoundingSphere(
+    geometry: THREE.BufferGeometry | DisplacedBufferGeometry,
+    featureBeginIndex: number,
+    featureEndIndex: number
+): THREE.Sphere {
+    let displacementRange: DisplacementRange | undefined;
+
+    if (geometry instanceof DisplacedBufferGeometry) {
+        displacementRange = geometry.displacementRange;
+        geometry = geometry.originalGeometry;
+    }
+
+    const attributes = geometry.attributes;
+    const pos = attributes.position as THREE.BufferAttribute;
+    const indices = geometry.index!.array;
+    const sphere = new THREE.Sphere();
+    const bbox = tmpBox.makeEmpty();
+    const vertex = tmpV1;
+
+    // First compute the bounding box for all line segments.
+    for (let i = featureBeginIndex; i < featureEndIndex; i += SEGMENT_STRIDE) {
+        bbox.expandByPoint(vertex.fromBufferAttribute(pos, indices[i]));
+        bbox.expandByPoint(vertex.fromBufferAttribute(pos, indices[i + VERTEX_STRIDE]));
+    }
+
+    const normal = tmpV2;
+    if (displacementRange) {
+        // If geometry is displaced, expand the bounding box to cover the whole displacement range,
+        // and return the sphere bounding the box. This is a coarse estimation, but avoids having
+        // to displace all vertices.
+        return expandBoxByDisplacementRange(
+            bbox,
+            displacementRange,
+            normal.fromBufferAttribute(geometry.attributes.normal as THREE.BufferAttribute, 0)
```

Comment:
True. For sphere projection the normals are not equal. This is assuming though the they're similar enough to not matter in practice. 
But it's true that the resulting bbox can be potentially wrong for big enough objects.

To correct this, we can expand the box in the two directions corresponding to the min and max normals in the geometry. Computing these directions would imply though going through all normals and I'd like to avoid that if possible. Another option would be to pass to the mesh constructor the min/max normals corresponding to the tile where the object is.

Reply:
I've been checking how bad can the approximation error be for different zoom levels by computing the dot product of the normals at the min max points of each tile. The results show that the error is very small starting from zoom level 4. See here some results:

TILE: (6,9,4), DOT: 0.9806892129880023
TILE: (12,17,5), DOT: 0.9946739445457075
TILE: (25,34,6), DOT: 0.9986326302953471
TILE: (50,68,7), DOT: 0.9996583822992287
TILE: (101,137,8), DOT: 0.9999152958483241
TILE: (202,274,9), DOT: 0.9999788289478592
TILE: (405,549,10), DOT: 0.9999947181695241
TILE: (810,1099,11), DOT: 0.9999986814690914
TILE: (1620,2199,12), DOT: 0.9999996706085572
...

Given this, I think the approximation is pretty valid. Trying to make it more accurate would make bbox computation slower unnecessarily.